src/test/java/com/google/devtools/build/lib/packages/TargetUtilsTest.java

```diff
@@ -96,4 +98,177 @@ public void testExecutionInfo() throws Exception {
     execInfo = TargetUtils.getExecutionInfo(tag1b);
     assertThat(execInfo).containsExactly("local", "", "cpu:4", "");
   }
+
+  @Test
+  public void testExecutionInfo_withPrefixSupports() throws Exception {
+    scratch.file(
+        "tests/BUILD",
+        "sh_binary(name = 'with-prefix-supports', srcs=['sh.sh'], tags=['supports-workers',"
+            + " 'supports-whatever', 'my-tag'])");
```

Comment:
Nit: scratch.file accepts varargs for the content, putting each string on a separate line, so this works:
scratch.file(
  "tests/BUILD",
  "sh_binary(",
  "  name = 'asdf',",
  "  srcs = [...],",
...)